src/av2/torch/dataloaders/utils.py

```diff
@@ -0,0 +1,293 @@
+
+        Raises:
+            NotImplementedError: If the cuboid mode conversion isn't supported.
+        """
+        if src == target:
+            return dataframe
+        if src == CuboidMode.XYZLWH_QWXYZ and target == CuboidMode.XYZLWH_THETA:
+            quaternions = dataframe.select(pl.col(list(QUAT_WXYZ_FIELDS))).to_numpy()
+            rotation = quat_to_mat(quaternions)
+            yaw = mat_to_xyz(rotation)[:, -1]
+
+            first_occurence = min(
+                i if field_name in QUAT_WXYZ_FIELDS else sys.maxsize
+                for (i, field_name) in enumerate(DEFAULT_ANNOTATIONS_TENSOR_FIELDS)
+            )
+            field_ordering = tuple(
+                filter(lambda field_name: field_name not in QUAT_WXYZ_FIELDS, DEFAULT_ANNOTATIONS_TENSOR_FIELDS)
+            )
+            field_ordering = field_ordering[:first_occurence] + ("yaw",) + field_ordering[first_occurence:]
+            dataframe = dataframe.with_columns(yaw=pl.from_numpy(yaw).to_series())
+            dataframe = dataframe.select(pl.col(list(field_ordering)))
+        elif src == CuboidMode.XYZLWH_QWXYZ and target == CuboidMode.XYZ:
+            unit_vertices_obj_xyz_m = np.array(
+                [
+                    [+1, +1, +1],  # 0
+                    [+1, -1, +1],  # 1
+                    [+1, -1, -1],  # 2
+                    [+1, +1, -1],  # 3
+                    [-1, +1, +1],  # 4
+                    [-1, -1, +1],  # 5
+                    [-1, -1, -1],  # 6
+                    [-1, +1, -1],  # 7
+                ],
+            )
+
+            dims_lwh_m = dataframe.select(pl.col(["length_m", "width_m", "height_m"])).to_numpy()
+
+            # Transform unit polygons.
+            vertices_obj_xyz_m = (dims_lwh_m[:, None] / 2.0) * unit_vertices_obj_xyz_m[None]
+
+            quat = dataframe.select(pl.col(list(QUAT_WXYZ_FIELDS))).to_numpy()
+            rotation = quat_to_mat(quat)
+            translation = dataframe.select(pl.col(["tx_m", "ty_m", "tz_m"])).to_numpy()
+
+            vertices = (rotation @ vertices_obj_xyz_m.transpose(0, 2, 1)).transpose(0, 2, 1) + translation[:, None]
+            columns = list(
+                itertools.chain.from_iterable(
+                    [(f"tx_{i}", f"ty_{i}", f"tz_{i}") for i in range(len(unit_vertices_obj_xyz_m))]
+                )
+            )
+            vertices = vertices.reshape(-1, len(unit_vertices_obj_xyz_m) * 3)
+            dataframe = pl.concat(
+                [
+                    dataframe.select(pl.col("*").exclude(["tx_m", "ty_m", "tz_m", "qw", "qx", "qy", "qz"])),
+                    pl.from_numpy(vertices, columns=columns, orient="row"),
+                ],
+                how="horizontal",
+            )
+            return dataframe
+        else:
+            raise NotImplementedError("This conversion is not implemented!")
+        return dataframe
+
+
+@dataclass(frozen=True)
+class Annotations:
+    """Dataclass for ground truth annotations."""
```

Comment:
[Nit] Might want to add a brief description for the dataclass attributes. Same for the dataclasses below.